api/app/src/command/medical/document/document-query.ts

```diff
@@ -14,38 +23,75 @@ export async function queryDocumentsAcrossHIEs({
   cxId: string;
   patientId: string;
   facilityId: string;
-}): Promise<DocumentQueryStatus> {
+}): Promise<DocumentQueryResp> {
   const patient = await getPatientOrFail({ id: patientId, cxId });
-  if (patient.data.documentQueryStatus === "processing") return "processing";
+  if (patient.data.documentQueryStatus === "processing")
+    return createQueryResponse("processing", patient);
 
   const externalData = patient.data.externalData?.COMMONWELL;
-  if (!externalData) return "completed";
+  if (!externalData) return createQueryResponse("completed", patient);
 
   const cwData = externalData as PatientDataCommonwell;
-  if (!cwData.patientId) return "completed";
+  if (!cwData.patientId) return createQueryResponse("completed", patient);
 
-  await updateDocQueryStatus({ patient, status: "processing" });
+  await updateDocQuery({ patient, status: "processing" });
 
   // intentionally asynchronous, not waiting for the result
   getDocumentsFromCW({ patient, facilityId }).catch(
     processAsyncError(`doc.list.getDocumentsFromCW`)
   );
 
-  return "processing";
+  return createQueryResponse("processing", patient);
 }
 
-export const updateDocQueryStatus = async ({
+const createQueryResponse = (status: DocumentQueryStatus, patient: Patient): DocumentQueryResp => {
+  return { queryStatus: status, queryProgress: patient.data.documentQueryProgress };
+};
+
+export const updateDocQuery = async ({
   patient,
   status,
+  progress,
 }: {
   patient: Patient;
   status: DocumentQueryStatus;
-}): Promise<Patient> => {
-  const patientModel = await getPatientOrFail({ id: patient.id, cxId: patient.cxId });
-  return patientModel.update({
-    data: {
-      ...patient.data,
-      documentQueryStatus: status,
-    },
-  });
+  progress?: {
+    completed: number;
+    total: number;
+  };
+}) => {
+  const sequelize = PatientModel.sequelize;
+  if (!sequelize) throw new Error("Missing sequelize");
+
+  let transaction: Transaction | undefined = await sequelize.transaction();
```

Comment:
this can also be null - perhaps we can just remove the explicit typing here? should be auto picked up by TS